File: app/serverless_v1.js
exports.handler = async (event, context) => {
    let count = 0 // initial state
  
    if (event.httpMethod === 'GET') {
      return {
        statusCode: 200,
        body: `Current count: ${count}`
      }
    } else if (event.httpMethod === 'POST') {
      const operation = event.path === '/increment' ? 'incremented' : 'decremented'
      count += operation === 'incremented' ? 1 : -1
      return {
        statusCode: 200,
        body: `Count ${operation} to ${count}`
      }
    } else {
      return {
        statusCode: 405,
        body: 'Method not allowed'
      }
    }
  }
  

File: httpserver/serverless_v1.py
import os
import boto3

s3 = boto3.resource('s3')

def lambda_handler(event, context):
    bucket_name = 'your-bucket-name'
    if event['httpMethod'] == 'GET':
        path = event['path']
        if path == '/':
            path = '/index.html'
        try:
            if 'html' in path:
                mimetype = 'text/html'
                object_key = path[1:]
                object = s3.Object(bucket_name, object_key)
                content = object.get()['Body'].read().decode('utf-8')
                response = {
                    'statusCode': 200,
                    'headers': {
                        'Content-Type': mimetype
                    },
                    'body': content
                }
                return response
            else:
                raise Exception('File not found')
        except Exception as e:
            response = {
                'statusCode': 404,
                'body': 'File not found'
            }
            return response
    else:
        response = {
            'statusCode': 405,
            'body': 'Method not allowed'
        }
        return response


